format-delimited/src/main/java/io/cdap/plugin/format/delimited/input/SplitQuotesIterator.java

```diff
@@ -61,6 +71,7 @@ protected String computeNext() {
       if (cur == QUOTE_CHAR) {
         isWithinQuotes = !isWithinQuotes;
         index++;
+        appendNextLineToDelimitedString(split, isWithinQuotes);
```

Comment:
nit: this is a bit confusing since it doesn't always append. let's rename it to appendNextLineIfNeeded()